test/controllers/owners_controller_test.rb

```diff
@@ -10,14 +10,14 @@ def setup
 
 
   def test_owner_creation
-     value = {name: Faker::Name.name[1..18], email:Faker::Internet.email, password: "Abcdef@1", address:Faker::Lorem.paragraph}
+     value = {name: Faker::Name.name[1..18], email:Faker::Internet.email, password: @owner.password, address:Faker::Lorem.paragraph}
      response =  post "/owners", value
      assert_equal(Owner.last.email, value[:email])
      assert_equal(response.status,302)
   end
 
   def test_owner_creation_notsuccess_withoutemail
-    value  = { name:Faker::Name.name[1..18] , email: "" , password:"Abcdef@1" , address:Faker::Lorem.paragraph}
+    value  = { name:Faker::Name.name[1..18] , email: "" , password:@owner.password , address:Faker::Lorem.paragraph}
     response = post  "/owners", value
     assert_equal(response.status,302)
   end
```

Comment:
def test_owner_email_check_present_in_db
    value = {email: @owner.email }
    response = post "/owners", value 
    assert_equal(response.status,302)
  end



here the value is checked in db, how come 302 status is returned?

Reply:
checking the value if present in db i am redirecting to the same page. so i have written assert_equal() statement for the redirection